uniffi_bindgen/src/templates/CallbackInterfaceTemplate.rs

```diff
@@ -0,0 +1,102 @@
+{#
+// For each callback interface in the UDL, we assume the caller has provided a corresponding
+// Rust `trait` with the declared methods. We provide the traits for sending it across the FFI.
+// If the caller's trait does not match the shape and types declared in the UDL then the rust
+// compiler will complain with a type error.
+// 
+// The generated proxy will implement `Drop`, `Send` and `Debug`.
+#}
+{% let trait_name = obj.name() -%}
+{% let trait_impl = format!("{}Proxy", trait_name) -%}
+{% let foreign_callback_holder = format!("foreign_callback_{}_holder", trait_name)|upper -%}
+
+// Register a foreign callback for getting across the FFI.
+static {{ foreign_callback_holder }}: std::sync::atomic::AtomicUsize = std::sync::atomic::AtomicUsize::new(0);
+
+#[no_mangle]
+pub extern "C" fn {{ obj.ffi_init_callback().name() }}(callback: uniffi::ForeignCallback) {
+    uniffi::set_foreign_callback(&{{ foreign_callback_holder }}, callback);
+}
+
+// Make an implementation which will shell out to the foreign language.
+#[derive(Debug)]
+struct {{ trait_impl }} {
+  handle: u64
+}
+
+// We need Send so we can stash the callbacks in objects we're sharing with foreign languages
+// i.e. in the handle map.
+// Prepared for our target trait to declare:
+//  `trait {{ trait_name }}: Send + std::fmt::Debug`
+unsafe impl Send for {{ trait_impl }} {}
+
+impl {{ trait_name }} for {{ trait_impl }} {
+    {%- for meth in obj.methods() %}
+
+    {#- Method declaration #}
+    fn {{ meth.name() -}}
+    ({% call rs::arg_list_decl_with_prefix("&self", meth) %})
+    {%- match meth.return_type() %}
+    {%- when Some with (return_type) %} -> {{ return_type|type_rs }}
+    {% else -%}
+    {%- endmatch -%} { 
+
+    {#- Method body #}
+        uniffi::deps::log::debug!("{{ obj.name() }}.{{ meth.name() }}");
+        let callback = uniffi::get_foreign_callback(&{{ foreign_callback_holder }}).unwrap();
+
+    {#- Packing args into a RustBuffer #}
+        {% if meth.arguments().len() == 0 -%}
+        let args_rbuf = uniffi::RustBuffer::new();
```

Comment:
nit: I think it would be cleaner not to special-case a zero-length arguments list here, and just let the `for arg in meth.arguments()` loop execute zero times. The end result will be exactly the same (`RustBuffer::new()` calls `RustBuffer::from_vec` with an empty vec under the hood).